stream_alert/rule_processor/rules_engine.py

```diff
@@ -458,30 +527,42 @@ def rule_analysis(self, record, rule, payload, alerts):
             alerts.append(alert)
 
     @staticmethod
-    def check_alerts_duplication(record, rule, alerts):
-        """The method to check if the record has been added to alerts list already.
+    def _is_equal(record1, record2):
+        """Check if two records are same excluding data normalization and threat intel information
 
-        The reason we need to do check alerts duplication is because original records
-        would be modified by inserting normalization or/and IOC information if there
-        exist. Threat Intel feature will process normalized records again and it
-        will result alert duplication.
+        Compare key set before comparing values for better performance.
+
+        Args:
+            record1/record2 (dict): The parsed log w/wo data normalization and
+            threat intel information.
+
+        Returns:
+            bool: Return True only when two records have same keys and values.
+        """
+        record1_keys = set(record1.keys()) - _IGNORE_KEYS
+        record2_keys = set(record2.keys()) - _IGNORE_KEYS
+
+        if record1_keys != record2_keys:
+            return False
+
+        return all(record1[key] == record2[key] for key in record1_keys)
```

Comment:
This only checks if all `record_1` key/values are present in `record2`, but it doesn't check the other way around.  There's also a built-in `cmp()` method that you could use, but you'd have to pop out the ignore keys prior to comparison

Reply:
You are right! I thought about this. Two lines above where comparing if two set of keys from `record_1` and `record_2` are same. If they are not same, return False immediately.  It prevents us to check the other way around.